worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,400 @@
+import os
+import typing
+import math
+import threading
+import pdb
+
+from typing import Dict, List, Set, Tuple, TextIO
+from BaseClasses import Item, MultiWorld, Location, Tutorial, ItemClassification
+from .Items import get_item_names_per_category, item_table, filler_items, trap_items
+from .Locations import get_locations
+from .Options import yoshi_options, get_option_value
+from .Regions import create_regions
+from .SetupGame import setup_gamevars
+from worlds.AutoWorld import World, WebWorld
+from .Client import YISNIClient
+from .Rom import LocalRom, patch_rom, get_base_rom_path, YIDeltaPatch, USHASH
+import Patch
+import settings
+
+class YISettings(settings.Group):
+    class RomFile(settings.SNESRomPath):
+        """File name of the Yoshi's Island 1.0 US rom"""
+        description = "Yoshi's Island ROM File"
+        copy_to = "Super Mario World 2 - Yoshi's Island (U).sfc"
+        md5s = [USHASH]
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+
+class YIWeb(WebWorld):
+    theme = "ocean"
+
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Yoshi's Island randomizer and connecting to an Archipelago server.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Pink Switch"]
+    )
+
+    tutorials = [setup_en]
+
+class YIWorld(World):
+    """Yoshi's Island is a 2D platforming game.
+        During a delivery, Bowser's evil ward, Kamek, attacked the stork, kidnapping Luigi and dropping Mario onto Yoshi's Island.
+        As Yoshi, you must run, jump, and throw eggs to escort the baby Mario across the island to defeat Bowser and reunite the two brothers with their parents."""
+    game: str = "Yoshi's Island"
+    option_definitions = yoshi_options
+    data_version = 1
+    required_client_version = (0, 3, 5)
```

Comment:
Wowie
Does the client still connect with 0.3.5? Probably should update this here, and there. Haha

Reply:
Maybe? I don't know if there would be any issues with it. SMW still has this as the minimum client, though.

Reply:
It can't possibly connect with an 0.3.5 client as that doesn't exist. minimum client version for initial implementation of a client derived from common client makes no sense. If one is defined at all it should be 0.4.4 to prevent early apworld clients from connecting.